### ISSUE TITLE
Roundtrip serialization of nested maps with empty value not working

### ISSUE THREAD

**Issue by morganmcc** [Kotlin/kotlinx.serialization#438]
I have a simple data class I am using to serialize into Redis. I have one field which is a nested string map like the following.

```kotlin
@SerialId(2) @Optional val groups: Map<String, Map<String, String>> = emptyMap()
```

I ran into an issue when this field contained a map that had an empty map as a value.

```
groups = mapOf("FOO" to emptyMap())
```

The object can be serialized without problem, but when you try to read the same object back and de-serialize it, you run into a missing required value exception. My guess is this is because the empty map is not serialized such that it can be brought back.

We got around it in our case by simply ignoring incoming data that would cause this problem, but  I could see it popping back up in other places.

**Comment by sandwwraith** [comment]
I think it depends on a serialization format which you use. Do you use default `Mapper` or a custom Encoder/Decoder?

**Comment by morganmcc** [comment]
My bad. I totally forgot to mention I was using the ProtoBuf serialization. Here is the actual model I was using. I can dump this with ProtoBuf.dump but ProtoBuf.load fails.

```kotlin
@Serializable
data class MonElementGroupData(
    @Transient val uuid: String = "",
    @SerialId(1) @Optional val version: Int = 1,
    @SerialId(2) @Optional val groups: Map<String, Map<String, String>> = emptyMap(),
    @SerialId(3) @Optional val lastUpdate: Long = 0
) 
```

**Comment by sandwwraith** [comment]
Well, according to this (https://github.com/protocolbuffers/protobuf/issues/4596), nested maps in protobuf are not supported. It is possible to implement it in kotlinx.serialization though, but I don't see much sense in it since it is a non-standard extension. I'd better add proper error message.

**Comment by morganmcc** [comment]
So it does. I totally missed that, thanks. It’s definitely totally reasonable to follow the Protobuf spec. With that in mind, I would suggest making both the serialization and deserialization operations fail with the same error, as right now the serialization call works.

**Comment by shanshin** [comment]
Fixed in 1.2.2 with 8f82c6b
